src/controllers/http/worker-http-controller.ts

```diff
@@ -0,0 +1,191 @@
+
+import { Worker } from "bullmq";
+import { Redis, Cluster } from "ioredis";
+import { debug } from "../../utils/log";
+
+import { HttpHandlerOpts, WorkerMetadata } from "../../interfaces";
+import { validateWorkerMetadata } from "../../validators";
+import { config } from "../../config";
+
+const debugEnabled = config.debugEnabled;
+
+const workers: { [queueName: string]: Worker } = {};
+
+const workerMetadataKey = config.workerMetadataKey;
+
+// Gracefully close all workers
+process.on('exit', async () => {
+  for (const queueName in workers) {
+    await workers[queueName].close();
+  }
+});
+
+const workerFromMetadata = (queueName: string, workerMetadata: WorkerMetadata, connection: Redis | Cluster): Worker => {
+  const { endpoint: workerEndpoint, opts: workerOptions } = workerMetadata;
+
+  debugEnabled && debug(`Starting worker for queue ${queueName} with endpoint ${workerMetadata.endpoint.url} and options ${workerMetadata.opts || 'default'}`);
+
+  const worker = new Worker(queueName, async (job) => {
+    debugEnabled && debug(`Processing job ${job.id} from queue ${queueName} with endpoint ${workerEndpoint.url}`);
+
+    // Process job by calling an external service using the worker endpoint
+    const controller = new AbortController()
+    const timeoutId = setTimeout(() => {
+      debugEnabled && ("Timeout, aborting...")
+      controller.abort()
+    }, workerEndpoint.timeout || 3000)
+
+    try {
+      const response = await fetch(workerEndpoint.url, {
```

Comment:
are you sure you don't want to use a lightweight wrapper like wretch to write less boilerplate? or even undici, which is also faster

Reply:
Not really. I prefer less dependencies actually unless the savings in code are really meaningful.